src/web/session_manager.cc

```diff
@@ -163,30 +163,33 @@ std::shared_ptr<Session> SessionManager::createSession(long timeout)
 
 std::shared_ptr<Session> SessionManager::getSession(const std::string& sessionID, bool doLock)
 {
+    if (sessions.empty()) {
+        return nullptr;
+    }
+
     std::unique_lock<decltype(mutex)> lock(mutex, std::defer_lock);
```

Comment:
decltype is not needed in C++17. just unique_lock, ctad should do the work.

Reply:
Removed.